backend/src/controllers/dish.ts

```diff
@@ -1,94 +1,68 @@
 import { Request, Response } from 'express'
 
-import { db } from '../services/firebase'
 import { Dish } from '../models/dish'
 import { Transaction } from '../models/transaction'
-import { mapDishesToLatestTransaction, mapToDishVM } from '../services/dish'
+import {
+    getAllDishes,
+    getAllUserDishes,
+    getAllUserDishesInUse,
+    getAllUserDishesVM,
+    mapDishesToLatestTransaction,
+    mapToDishVM,
+} from '../services/dish'
 import { CustomRequest } from '../middlewares/auth'
+import { verifyIfUserAdmin } from '../services/users'
+import { getAllTransactions } from '../services/transactions'
 import Logger from '../utils/logger'
 
 export const getDishes = async (req: Request, res: Response) => {
-    // TODO: send dish information based on user role
+    let userClaims = (req as CustomRequest).firebase
+    let all = req.query['all']?.toString()
+    let withTransactions = req.query['transaction']?.toString()
+    let borrowed = req.query['borrowed']?.toString()
 
-    // Example of how to get user claims
-
-    // let userClaims = (req as CustomRequest).firebase
-    // userClaims.role
-
-    // get dishes from firebase
-    let rawDishData = <Array<Dish>>[]
+    let allDishes = <Array<Dish>>[]
     try {
-        let dishesQuerySnapshot = await db.collection('dishes').get()
-        dishesQuerySnapshot.docs.forEach((doc) => {
-            let data = doc.data()
-            rawDishData.push({
-                id: doc.id,
-                qid: parseInt(data.qid, 10),
-                registered: data.registered.toDate(),
-                type: data.type ? data.type : '',
-            })
+        allDishes = await getAllDishes()
```

Comment:
I don't want the logic to fetch all the transactions from the Firestore database as if the app grows there can be a hundred thousand transactions that will be fetched before any dish borrow logic is even used. This will severely impact the response time as we are using an extremely slow free version of Firebase. Fetching and parsing this much data can really intensive.
<hr>

1. I want to only fetch data from Firebase with conditions only when we require it. This will help us batch the Firebase requests into smaller chunks improving response time. 
2. We can talk over the design on how to go about breaking the transaction fetch logic from Firebase to improve it.
3. We might need to add a caching layer to our API calls in future to reduce the API call made to Firestore for repetitive same data.